### PR TITLE
#3 Build Quiz

### PR DESCRIPTION
**Pull request by rheien**
create question cards for building a quiz

### CODE REVIEW

File: src/questions/singleChoiceQuestions.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export const singleChoiceQuestions : Question[]= [
+    
+    {
+        question : "Wie schreibt man",
+        answers : [
+             'Portemonnaie',
+             'Portmonnaie',
+             'Portemonaie'
+        ], 
+        correctAnswers : ['a']
```
Comment (annamarie-now) [inline src/questions/singleChoiceQuestions.ts:11]: Die korrekte Antwort ist a als String? Was ist a?

File: src/components/QuizBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        return picks;
+    }
+
+    /* get random number for picking a question */
+    randomPick(questions: Question[]) : number{
+        return Math.floor(Math.random() * questions.length)
+    }
+
+    /* shuffle function */
+    shuffle(array : any[]) {
```
Comment (annamarie-now) [inline src/components/QuizBuilder.ts:50]: Der Funktionsname "shuffle" ist aus meiner Sicht auch zu unpräzise (siehe vorherigen Kommentar)

File: src/components/QuizBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                pickNumber = this.randomPick(typeOfAnswers);
+            }
+
+            picks.push(typeOfAnswers[pickNumber]);
+        }
+        return picks;
+    }
+
+    /* get random number for picking a question */
+    randomPick(questions: Question[]) : number{
```
Comment (annamarie-now) [inline src/components/QuizBuilder.ts:45]: Gibt es einen präziseren Funktionsnamen?

**Review by annamarie-now — CHANGES_REQUESTED** [review]

File: src/questions/singleChoiceQuestions.ts (hunk 1 above)
Comment (rheien) [reply src/questions/singleChoiceQuestions.ts:11]: das wurde schon geändert

File: src/components/QuizBuilder.ts (hunk 3 above)
Comment (rheien) [reply src/components/QuizBuilder.ts]: hoffe die Benennung passt

File: src/components/QuizBuilder.ts (hunk 2 above)
Comment (rheien) [reply src/components/QuizBuilder.ts]: hab den Namen spezifiziert

File: src/questions/singleChoiceQuestions.ts (hunk 1 above)
Comment (rheien) [reply src/questions/singleChoiceQuestions.ts]: hatten es schon verändert. nur vergessen zu pushen

File: src/components/QuizBuilder.ts [hunk 4]
```diff
@@ -42,12 +42,12 @@ export class QuizBuilder {
...
 
     /* get random number for picking a question */
-    randomPick(questions: Question[]) : number{
+    randomPickQuestion(questions: Question[]) : number{
         return Math.floor(Math.random() * questions.length)
     }
 
     /* shuffle function */
-    shuffle(array : any[]) {
+    shuffleQuestionsOrder(array : any[]) {
```
Comment (querwurzel) [inline src/components/QuizBuilder.ts:50]: This method has zero knowledge about questions and what they are .. it takes _any_ array and shuffles .. which leads to another point:

Is this method really part of a quiz builder?

File: src/components/QuizBuilder.ts [hunk 5]
```diff
@@ -22,18 +22,18 @@ export class QuizBuilder {
...
+        quiz.questions = this.shuffleQuestionsOrder(questions);
         return quiz;
     };
 
     /* pick questions for the quiz */
     poseQuestion(typeOfAnswers: Question[]) : Question[]{
         const picks : Question[] = [];
         for (let index = 0; index < 2; index++) {
-            let pickNumber: number = this.randomPick(typeOfAnswers);
+            let pickNumber: number = this.randomPickQuestion(typeOfAnswers);
```
Comment (querwurzel) [inline src/components/QuizBuilder.ts:33]: What do you think of the readability of this method name?

File: src/components/QuizBuilder.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+        questions = questions.concat(this.poseQuestion(multipleChoiceQuestions));
+        questions = questions.concat(this.poseQuestion(singleChoiceQuestions));
+        questions = questions.concat(this.poseQuestion(textInputQuestions));
+        
+        quiz.questions = this.shuffle(questions);
+        return quiz;
+    };
+
+    /* pick questions for the quiz */
```
Comment (querwurzel) [inline src/components/QuizBuilder.ts:29]: Although I like you describing your methods I think you should have a quick look into https://jsdoc.app/about-getting-started.html .. you will find a convention for documention

File: src/components/QuizBuilder.ts (hunk 5 above)
Comment (rheien) [reply src/components/QuizBuilder.ts:33]: oh no... the intention of my method is to describe something else

File: src/components/QuizBuilder.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        questions = questions.concat(this.poseQuestion(multipleChoiceQuestions));
+        questions = questions.concat(this.poseQuestion(singleChoiceQuestions));
+        questions = questions.concat(this.poseQuestion(textInputQuestions));
+        
+        quiz.questions = this.shuffleQuestionsOrder(questions);
+        return quiz;
+    };
+
+    /* pick questions for the quiz */
+    poseQuestion(typeOfAnswers: Question[]) : Question[]{
```
[the remark this answers is not in the document]
Comment (rheien) [reply src/components/QuizBuilder.ts:30]: oh yea plural

File: src/components/QuizBuilder.ts (hunk 4 above)
Comment (rheien) [reply src/components/QuizBuilder.ts:50]: no it doesn't belong. i wanted to use it later for the answers too

Comment (querwurzel) [reply src/components/QuizBuilder.ts:50]: You could leave it for now and make it a static method .. have a look about static methods (same as in Java)

**Review by annamarie-now — APPROVED** [review]

**Comment by annamarie-now** [comment]
Nice!

**Pull request merged**